packages/aws-cdk-lib/aws-s3/lib/bucket.ts

```diff
@@ -2424,7 +2424,7 @@ export class Bucket extends BucketBase {
     return this.inventories.map((inventory, index) => {
       const format = inventory.format ?? InventoryFormat.CSV;
       const frequency = inventory.frequency ?? InventoryFrequency.WEEKLY;
-      const id = inventory.inventoryId ?? `${this.node.id}Inventory${index}`;
+      const id = inventory.inventoryId ?? `${this.node.id}Inventory${index}`.replace( /[^\w\.\-]/g, '').slice(-64);
```

Comment:
Are these conditions in the docs somewhere? I don't see them [here](https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutBucketInventoryConfiguration.html).

Reply:
Correct, @msambol they are not documented. That's why I added the conversation with the AWS Premium Support above.